std/io.zig

```diff
@@ -245,6 +263,18 @@ pub const File = struct {
                 }
                 return result;
             },
+            Os.windows => {
+                var pos : usize = undefined;
+                if (system.SetFilePointerEx(self.handle, 0, @ptrCast(system.PLARGE_INTEGER, &pos), system.FILE_CURRENT) == 0) {
```

Comment:
`LARGE_INTEGER` is always an `i64`. `usize` is pointer-sized. These are not necessarily the same byte size. Also the signed-ness is different. So part of the job of this function is to get the i64 from Windows, and then convert it to a `usize`, making correct decisions about how to handle the following situations:
 * when usize is smaller than i64 (probably return `error.FilePosLargerThanPointerRange`, which is maybe a hint that we should actually return u64 instead of usize from getPos, but that's a separate issue, which could be fixed later)
 * when the i64 we get from microsoft is negative instead of positive (maybe we assert that it will not happen? maybe we return an error?)

Generally, `@ptrCast` should be avoided if possible, and this is a situation where I don't think we need one.